src/zenml/integrations/aws/step_operators/sagemaker_step_operator.py

```diff
@@ -159,20 +159,37 @@ def launch(
 
         settings = cast(SagemakerStepOperatorSettings, self.get_settings(info))
 
+        # Get and default fill SageMaker estimator arguments for full ZenML support
+        estimator_args = settings.estimator_args
         session = sagemaker.Session(default_bucket=self.config.bucket)
-        instance_type = settings.instance_type or "ml.m5.large"
+
+        if "instance_type" not in estimator_args:
```

Comment:
Not necessary to get this approved, but I'll point it out anyway because I think it's a little neater: There is a dict method called `setdefault()` which does the same thing:
```python
estimator_args.setdefault("instance_type", "ml.m5.large")
```

Reply:
To keep backwards compatibility with the existing settings:
```python
estimator_args.setdefault("instance_type", settings.instance_type or "ml.m5.large")
```

Reply:
Added.